src/js/components/TabPanesComponent.jsx

```diff
@@ -56,6 +56,12 @@ var TabPanesComponent = React.createClass({
               className="btn btn-success pull-right">
                   + New App
             </Link>
+            <Link
+                to={path}
+                query={{modal: "chaos"}}
+                className="btn btn-default pull-right">
+                  Chaos
```

Comment:
Also see my comment below.
makeChaos() is an asynchronos operation what can take a lot of time if you have a large amount of apps, where all the tasks needs collected. During this time the button should be disabled.